flyteidl/protos/flyteidl/admin/agent.proto

```diff
@@ -35,46 +40,79 @@ message TaskExecutionMetadata {
   string k8s_service_account = 5;
   // Environment variables attached to the task execution
   map<string, string> environment_variables = 6;
+  int32 max_attempts = 7;
+  bool interruptible = 8;
+  int32 interruptible_failure_threshold = 9;
+  TaskOverrides overrides = 10;
 }
 
 // Represents a request structure to create task.
 message CreateTaskRequest {
-  // The inputs required to start the execution. All required inputs must be
-  // included in this map. If not required and not provided, defaults apply.
-  // +optional
-  core.LiteralMap inputs = 1;
+  oneof part {
```

Comment:
this feels odd and does not seem to have a clear understanding of what will be sent. 
Inputs and header should always be sent i feel.

IMO, CreateTask should remain sync, we can add a ExecuteSyncStream that sounds like a better option if we need stream